main.c

```diff
@@ -556,6 +556,60 @@ exiterr (GError *error)
   exit (EXIT_FAILURE);
 }
 
+/**
+ * apply_excludes:
+ * @pool: initialized pool
+ * @excludes: (out): a map of solvables
+ * @exclude_packages: names of packages to be excluded
+ * @lookasides: set of repos that will not have packages excluded
+ * @modular_pkgs: map of modular packages that exclusion does not apply to
+ *
+ * Apply excludes: this completely hides the package from any processing.
+ * Packages in lookaside are not removed, and neither are modular packages.
+ */
+static void
+apply_excludes(Pool *pool,
+               Map *excludes,
+               GStrv exclude_packages,
+               GHashTable *lookaside_repos,
+               Map *modular_pkgs)
+{
+    map_init (excludes, pool->nsolvables);
+    /* This map has 1 for all available packages, 0 for excluded ones. */
+    map_setall (excludes);
+
+    for (GStrv exclude = exclude_packages; exclude && *exclude; exclude++) {
+        g_auto(Queue) sel;
+        queue_init (&sel);
+        selection_make (pool, &sel, *exclude, SELECTION_NAME | SELECTION_DOTARCH);
+        if (!sel.count) {
+            g_warning ("Nothing matches exclude '%s'", *exclude);
+            continue;
+        }
+
+        g_auto(Queue) q;
+        queue_init (&q);
+        selection_solvables (pool, &sel, &q);
+
+        for (int j = 0; j < q.count; j++) {
+            Id p = q.elements[j];
+            Solvable *s = pool_id2solvable (pool, p);
+            if (g_hash_table_contains (lookaside_repos, s->repo)) {
+                /* Ignore packages from lookaside. */
+                continue;
+            }
+
+            if (map_tst (modular_pkgs, p)) {
+                /* Modular package, not excluding... */
```

Comment:
Are you sure? Also use g_debug because it will be easier to find why package is still there

Reply:
I think this matches what @contyk wrote in #3. Modular packages should be excluded by filter_packages in modulemd, and they will not be listed in the artifacts at all. I think the filtered packages will not even be included in the source repos.

If a module includes a package incorrectly, it will have to be rebuilt to remove that package.